pkg/controllers/work/apply_controller.go

```diff
@@ -531,6 +562,7 @@ func computeManifestHash(obj *unstructured.Unstructured) (string, error) {
 	manifest.SetSelfLink("")
 	manifest.SetDeletionTimestamp(nil)
 	manifest.SetManagedFields(nil)
+	manifest.SetOwnerReferences(nil)
```

Comment:
Had to make this change to get E2E to pass for spec hash check, because spec hash for object kept changing on different due live UID field in OwnerRef which changes only on different E2E runs.

Reply:
This causes some UTs to fail to fix it we need to remove live fields UID, BlockOwnerDeletion and Controller field if populated in OwnerRef 
![Uploading image.png…]()

Reply:
reverted change and added comment explaining why